cmd/generate.go

```diff
@@ -0,0 +1,171 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+)
+
+// generateCmd represents the generate command
+var generateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "Generate docker-compose.yml and harbor-compose.yml files from an existing shipment",
+	Long: `
+$ harbor-compose generate my-shipment dev
+or
+$ harbor-compose generate my-shipment qa
+`,
+	Run: generate,
+}
+
+func init() {
+	RootCmd.AddCommand(generateCmd)
+}
+
+func generate(cmd *cobra.Command, args []string) {
+	if len(args) < 2 {
+		log.Fatal("2 arguments are required. ex: harbor-compose generate my-shipment dev")
+	}
+
+	shipment := args[0]
+	env := args[1]
+
+	if Verbose {
+		log.Printf("fetching shipment...")
+	}
+	shipmentObject := GetShipmentEnvironment(shipment, env)
+
+	//convert a Shipment object into a DockerCompose object
+	dockerCompose := DockerCompose{
+		Version:  "2",
+		Services: make(map[string]DockerComposeService),
+	}
+
+	//convert containers to docker services
+	for _, container := range shipmentObject.Containers {
+
+		//create a docker service based on this container
+		service := DockerComposeService{
+			Image:       container.Image,
+			Ports:       make([]string, len(shipmentObject.Ports)),
+			Environment: make(map[string]string),
+		}
+
+		//populate ports
+		for _, port := range container.Ports {
+
+			//format = external:internal
+			if port.PublicPort == 0 {
+				port.PublicPort = 80
```

Comment:
so are you setting a PublicPort for them? This should be allowed to be empty and instead use the PORT value.